server/dto/DTODecoder.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type-Reborn
+** File description:
+** DTODecoder.hpp
+*/
+
+#ifndef DTODECODER_HPP
+#define DTODECODER_HPP
+
+#include "IDTODecoder.hpp"
+#include "DTORegistry.hpp"
+#include "DTOException.hpp"
+
+/**
+ * @class DTODecoder
+ * @brief Class for DTO decoder
+ * @inherit IDTODecoder
+ * @version v0.1.0
+ * @since v0.1.0
+ * @author Marius PAIN
+ */
+class DTODecoder final: public IDTODecoder {
+public:
+	/**
+     * @brief The constructor of the DTODecoder class
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Marius PAIN
+     */
+	DTODecoder();
+
+	/**
+     * @brief The constructor of the DTODecoder class
+     * @param dtoRegistry The DTO registry
+     * @version v0.1.0
+     * @since v0.1.0
+     * @author Marius PAIN
+     */
+	explicit DTODecoder(DTORegistry *dtoRegistry);
```

Comment:
```suggestion
	explicit DTODecoder(DTORegistry *dtoRegistry = nullptr);
```
This simplifies your class because it can have only have 1 constructor